src/_guides/language/language-tour.md

```diff
@@ -3862,9 +3863,30 @@ print(names is List<String>); // true
 ### Restricting the parameterized type
 
 When implementing a generic type,
-you might want to limit the types of its parameters.
+you might want to limit the types of its parameters
+to be a subtype of a particular type.
 You can do this using `extends`.
 
+A common use case is ensuring that a type is non-nullable
+by making it a subtype of `Object`
+(instead of the default, [`Object?`][top-and-bottom]).
+
+<?code-excerpt "misc/lib/language_tour/generics/misc.dart (non-nullable)"?>
+```dart
+class Foo<T extends Object> {
+  // Foo works on objects with the non-nullable type T.
+}
+```
+
+How does `Foo<T extends Object>` differ from `Foo<Object>`?
+If you use `<Object>`, then objects used in
+`Foo` members can have _any_ subtype of `Object`.
+Adding `T extends` lets you restrict the type
+to subtypes of `T`.
+
+You can use `extends` with other types besides `Object`.
```

Comment:
Maybe worth pointing out *why* you would do this?  The main use case is that it allows members of `SomeBaseClass` to be called on things of type `T`.